### ISSUE TITLE
Incorrect Answer and Wrong Description of Gradient Descent

### ISSUE THREAD

**Issue by mindful-math** [Open-Deep-ML/DML-OpenProblem#97]
The problem "Linear Regression Using Gradient Descent (easy)" is wrong and poorly worded. Gradient descent is given by the following iterative procedure...

$$\theta_t \leftarrow \theta_{t-1} - \gamma \nabla f(\theta_{t-1})$$

where $f(\cdot)$ is convex, $\gamma$ denotes the learning rate, and $\nabla f(\cdot)$ denotes the gradient of whatever function it is you are trying to minimize. In the case of linear regression, our loss is given by 

$$ \mathcal{L}(\hat{\theta}) = \frac{1}{m}\lVert y-X\hat{\theta} \rVert_2^2$$

where the gradient is given by 

$$\frac{2}{m}\left( X^T (X\hat{\theta} - y) \right)$$

and so our update is

$$\theta_t \leftarrow \theta_{t-1} - \frac{2\gamma}{m} \left( X^T (X\theta_{t-1} - y) \right)$$

not this mystical stuff about the hypothesis minus the actual value times the data (note the missing factor of two from the solution)... I'd also like to see a problem about implementing line search or any other step size regime (because you're trivializing the step size here with some constant for all iterates).

**Comment by Open-Deep-ML** [comment]
Thanks for the feedback. We want to clarify that the implementation is correct. The cost function used includes a factor of 1/2, which is why there’s no need for the extra factor of 2 in the gradient. This is standard practice in many gradient descent implementations for linear regression.

That being said, we can see how the problem statement could be made a bit clearer, and we will work on improving that. If you have any other concerns, please state them more politely. We always open to constructive criticism, but the solution here is sound.

If you'd like to suggest a new problem (e.g., line search or advanced step size methods), feel free to create an issue with the details. 

If you still have an issue with the learn section or solution, put in a PR https://github.com/Open-Deep-ML/DML-OpenProblem/tree/main/Problems/15_linear_regression_gradient_descent
